spec/lib/safe_commit_spec.rb

```diff
@@ -0,0 +1,151 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+
+RSpec.describe SafeCommit do
+  it "has a version number" do
+    expect(SafeCommit::VERSION).to_not be_nil
+  end
+
+  describe "#expect" do
+    it "returns an instance of Assertion with the guess set" do
+      guess = "some guess"
+      expect_result = described_class.expect(guess)
+      expect(expect_result).to be_instance_of(SafeCommit::Assertion)
+      expect(expect_result.guess).to eq(guess)
+    end
+  end
+
+  describe "#test_engine" do
+    it "sets the test engine for ModiFile instance" do
+      test_engine_setting = "rspec"
+      test_engine(test_engine_setting)
+      expect(ModiFile.instance.test_engine).to eq(test_engine_setting)
+    end
+  end
+
+  describe "#tests", skip: "unable to stub the `tests` call" do
+    let(:options) { { safety: true, verbose: false } }
+
+    it "returns '0 failures' when no errors occur during testing" do
+      allow(self).to receive(:extract_failed_tests_count).and_return("0 failures")
+      expect(tests(options)).to eq("0 failures")
+    end
+
+    it "returns 'Error running tests, skipping...' when errors occur during testing" do
+      allow(self).to receive(:run_rspec).and_return("errors occurred")
+      expect(tests(options)).to eq("Error running tests, skipping...")
+    end
+  end
```

Comment:
This isn't working because you aren't mocking `run_rspec` in the first `it` block. Managed to get it working like so:

```suggestion
  describe "#tests" do
    let(:options) { { safety: true, verbose: false } }

    it "returns '0 failures' when no errors occur during testing" do
      allow(self).to receive(:run_rspec).and_return("0 failures")
      allow(self).to receive(:extract_failed_tests_count).and_return("0 failures")
      expect(tests(options)).to eq("0 failures")
    end

    it "returns 'Error running tests, skipping...' when errors occur during testing" do
      allow(self).to receive(:run_rspec).and_return("errors occurred")
      expect(tests(options)).to eq("Error running tests, skipping...")
    end
  end
```